src/module-vsbridge-indexer-catalog/Model/Indexer/DataProvider/Product/PriceData.php

```diff
@@ -61,6 +61,11 @@ public function addData(array $indexData, $storeId)
             if (isset($priceDataRow['price'])) {
                 $indexData[$productId]['regular_price'] = $this->preparePrice($priceDataRow['price']);
             }
+
+            if ($this->preparePrice($indexData[$productId]['regular_price']) >= $this->preparePrice($indexData[$productId]['final_price'])) {
```

Comment:
no need to call `preparePrice` function on 'regular_price' - we already have here a float value: 
```
 if (isset($priceDataRow['price'])) {
                $indexData[$productId]['regular_price'] = $this->preparePrice($priceDataRow['price']);
            }
```
also in some cases `regular_price` might not be available.